staging/src/k8s.io/kubectl/pkg/scale/scale.go

```diff
@@ -130,7 +131,21 @@ func (s *genericScaler) ScaleSimple(namespace, name string, preconditions *Scale
 		return updatedScale.ResourceVersion, nil
 	}
 
-	patch := []byte(fmt.Sprintf(`{"spec":{"replicas":%d}}`, newSize))
+	type objectForReplicas = struct {
```

Comment:
how do anonymous structs interact with the type encoder cache?

https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L380

I have a slight preference for actually declaring the type as a private package type

Reply:
Learn something new, Thanks @liggitt 😄

Reply:
It interacts as expected. Deceleration of a type in a function only controls the visibility of the type.